src/Microsoft.OData.Core/UriParser/Parsers/FunctionCallParser.cs

```diff
@@ -60,11 +77,15 @@ public ExpressionLexer Lexer
         /// Parses an identifier that represents a function.
         /// </summary>
         /// <param name="parent">Token for the parent of the function being parsed.</param>
-        /// <returns>QueryToken representing this function.</returns>
-        public QueryToken ParseIdentifierAsFunction(QueryToken parent)
+        /// <param name="result">QueryToken representing this function.</param>
+        /// <returns>True if the parsing was successful.</returns>
+        public bool ParseIdentifierAsFunction(QueryToken parent, out QueryToken result)
         {
+            result = null;
             string functionName;
 
+            lexer.TakeTextPositionSnapshot();

```

Comment:
It would be great if you could implement this function without modifying the lexer.  Why is a snapshot needed anyway? 

If you can't find a way to do that, then lexer should probably support Snapshot() and Rollback().  If it can only support a single rollback, it should assert if it is attempted a second time.  A rollback without a Snapshot() can ignore the call (like it does).